// ngx-splide/esm2015/lib/ngx-splide.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxSplideComponent } from './ngx-splide.component';
import { NgxSplideSlideComponent } from './ngx-splide-slide.component';
import * as i0 from "@angular/core";
export class NgxSplideModule {
}
NgxSplideModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: NgxSplideModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxSplideModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: NgxSplideModule, declarations: [NgxSplideComponent, NgxSplideSlideComponent], imports: [CommonModule], exports: [NgxSplideComponent, NgxSplideSlideComponent] });
NgxSplideModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: NgxSplideModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: NgxSplideModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgxSplideComponent, NgxSplideSlideComponent],
                    imports: [CommonModule],
                    exports: [NgxSplideComponent, NgxSplideSlideComponent]
                }]
        }] });
